### DISCUSSION TITLE
Reflashing SNES cartmods

### DISCUSSION THREAD

**Discussion by Chomemel** [sanni/cartreader#839] [Need help]
I wonder if I can reprogramm SNES cartmods with 29LVxxx EEPROM.

I have different PCBs, but for each one, WE and WP are connected together and go to Vcc. Here a picture of one of them.
![unikenny](https://github.com/sanni/cartreader/assets/107198559/510b0a23-8d1c-4206-af24-4329cec38748)

I understood that when WE gets VIH (Vcc), writing process can’t occur. You need to be connected it to WR of the cart reader/console. 
I can’t connect only WE to WR without lifting the EEPROM’s pin (to disconnect WE from WP). 
**Can I flash the EEPROM if both WE and WP are connected to WR ?**

To be honnest I read the 29LV640 datasheet a lot, but I dont really understand how WP really works.

**Comment by sanni** [comment]
It's probably best to ask the seller of those PCBs how he flashes the chips.
The datasheet says WP needs to be high and WE needs to be low for flashing.

**Comment by Chomemel** [comment]
These PCB are open source and the ones not one the picture was a single batch and the "author" has become hard to reach.

But now that you make me think about it, I might know the answer (or partially).
It's possible to flash the chip after it is soldered like is this photo.
![IMG_20210116_164802](https://github.com/sanni/cartreader/assets/107198559/57e4b3de-b566-4a66-8b0d-7a35caa2c2c9)


**So if it works with a programmer, I guess it should work with the OSCR, right?**

**Comment by Gilberto99** [comment]
Could someone please provide me with GERBERs for the mx29lv320 adapter? I need it to make snes retros

**Reply by Chomemel** [reply]
It seems that the author removed it from from his oshpark shared projects. I don't have them sorry. But there are probably some from other authors you could find on the internet.

**Comment by Chomemel** [comment]
I realize I never updated my progress. I managed to reprogram these (29LVxxx) as well as 29L3211 (SOP44) and more recently 39SF040 directly on the cartridge. Here are pictures.

![IMG_20240401_000254](https://github.com/user-attachments/assets/e13c198c-fd9e-46a0-9a55-87e7566caa3f)
![IMG_20240502_231809](https://github.com/user-attachments/assets/312477c8-a568-41bc-b94c-8a220eb3812a)
![IMG_20240502_232237](https://github.com/user-attachments/assets/4e29ec20-ef8b-4cc3-9d29-185d44562d08)

Thanks again for that woderful project. When I built it, I was only interseted in dumping my saves. But I spent so much time doing other things with it.
